sycl/test-e2e/Basic/built-ins/marray_math.cpp

```diff
@@ -115,8 +117,33 @@ int main() {
   TEST3(sycl::nan, float, 3, ma7);
   if (deviceQueue.get_device().has(sycl::aspect::fp64))
     TEST3(sycl::nan, double, 3, ma8);
-  TEST(sycl::half_precision::exp10, float, 2, EXPECTED(float, 10, 100), 0.1,
+
+  TEST(sycl::half_precision::sin, float, 3,
+       EXPECTED(float, 0.98545f, -0.871576f, -0.832267f), 0.001, ma6);
+  TEST(sycl::half_precision::cos, float, 3,
+       EXPECTED(float, 0.169967, -0.490261, 0.554375), 0.001, ma6);
+  TEST(sycl::half_precision::tan, float, 3,
+       EXPECTED(float, 5.797, 1.777, -1.501), 0.001, ma6);
+  TEST(sycl::half_precision::divide, float, 2, EXPECTED(float, 3.0f, 1.0f), 0,
+       ma2, ma1);
+  TEST(sycl::half_precision::log, float, 2, EXPECTED(float, 0.0f, 0.693f),
+       0.001, ma1);
+  TEST(sycl::half_precision::log2, float, 2, EXPECTED(float, 0.0f, 1.f), 0,
        ma1);
+  TEST(sycl::half_precision::log10, float, 2, EXPECTED(float, 0.0f, 0.301f),
+       0.001, ma1);
+  TEST(sycl::half_precision::powr, float, 2, EXPECTED(float, 3.0f, 4.0f), 0,
```

Comment:
Optimally we would actually check the ULP instead of some random error border. I can see what I can do, but it might take a couple of days before I have a chance.

Reply:
When there are already examples of ULP check, please let us know. Thanks for the suggestion.

Reply:
I updated the tests for half precision with ULP check. Thanks.